### ISSUE TITLE
Successor #3807: verify CUDA training smoke after CuBLAS setup STOP

### ISSUE THREAD

**Issue by Unjuno** [Unjuno/agent-interface#3812]
Successor to the immutable preflight STOP on parent #3807. Do not amend or reinterpret that run. New allocation ID: `cuda-smoke-3807-successor-01`.

## H / T / D / C / U

**H**: With the documented CuBLAS determinism environment set before Python starts, the already-installed RTX 3080/PyTorch stack can complete 100 deterministic optimizer updates and improve held-out accuracy on a small synthetic binary task.

**T**: Execute the exact inline source below once, with process-scoped `CUBLAS_WORKSPACE_CONFIG=:4096:8`, `PYTHONDONTWRITEBYTECODE=1`, and Python 3.11.9 / PyTorch 2.5.1+cu121. Seed 3807; 1,024 training + 256 held-out rows; 8 features; fixed linear label function; Adam lr 0.05; exactly 100 full-batch steps. Verify source SHA-256 before invocation. Capture the single stdout/error directly in this issue. No retries.

**D**: `PASS_CUDA_TRAINING_SMOKE` iff train BCE decreases and held-out accuracy rises to at least 0.80; otherwise preserve exact STOP/FAIL. Any source hash, device, or environment mismatch is STOP.

**C**: RTX 3080 Laptop GPU only; deterministic algorithms on, TF32 off; no Docker (cached image reads currently fail), network, checkpoints, local files, dependency installation, pulls, cleanup, or runtime changes. C: has 0 free bytes; this allocation writes no host files and may proceed only as an in-memory preflight. This does not consume parent #3807's five-seed rank-4 allocation.

**U**: Small synthetic kernel/optimizer feasibility only; not LoRA/rank capacity, online feedback, route safety, runtime/product evidence, or an efficiency result.

Frozen source (UTF-8 without trailing newline), SHA-256 `3f543c969f56a3ad030112afd35c865b6143109bfbfd9526f16026ec96a03c3b`:

```python
import torch,json;torch.manual_seed(3807);torch.use_deterministic_algorithms(True);torch.backends.cuda.matmul.allow_tf32=False;torch.backends.cudnn.allow_tf32=False;g=torch.Generator().manual_seed(3807);x=torch.randn((1280,8),generator=g);w=torch.tensor([0.7,-0.4,0.2,0.0,0.0,0.0,0.0,0.0]);y=((x@w+0.15)>0).float().unsqueeze(1);x=x.cuda();y=y.cuda();model=torch.nn.Linear(8,1).cuda();opt=torch.optim.Adam(model.parameters(),lr=0.05);loss_fn=torch.nn.BCEWithLogitsLoss();before=float(loss_fn(model(x[:1024]),y[:1024]).item());initial=float((model(x[1024:]).squeeze(1).gt(0)==y[1024:].squeeze(1).bool()).float().mean().item());history=[];[(opt.zero_grad(set_to_none=True),loss_fn(model(x[:1024]),y[:1024]).backward(),opt.step()) for _ in range(100)];trainloss=float(loss_fn(model(x[:1024]),y[:1024]).item());pred=model(x[1024:]).squeeze(1).gt(0);acc=float((pred==y[1024:].squeeze(1).bool()).float().mean().item());print(json.dumps({"device":torch.cuda.get_device_name(0),"seed":3807,"steps":100,"train_loss_before":before,"train_loss_after":trainloss,"heldout_accuracy_before":initial,"heldout_accuracy_after":acc,"cuda":torch.version.cuda},sort_keys=True))
```

## Formal smoke outcome — 2026-09-21

**PASS_CUDA_TRAINING_SMOKE**. The inline UTF-8 source SHA-256 matched the frozen `3f543c969f56a3ad030112afd35c865b6143109bfbfd9526f16026ec96a03c3b` before the sole invocation. Local Windows Python 3.11.9 / PyTorch 2.5.1+cu121 / CUDA 12.1, NVIDIA GeForce RTX 3080 Laptop GPU; `CUBLAS_WORKSPACE_CONFIG=:4096:8`, `PYTHONDONTWRITEBYTECODE=1`, no Docker, checkpoints, local result files, installs, pulls, or cleanup.

Exact stdout: `{"cuda":"12.1","device":"NVIDIA GeForce RTX 3080 Laptop GPU","heldout_accuracy_after":0.9921875,"heldout_accuracy_before":0.5859375,"seed":3807,"steps":100,"train_loss_after":0.14097687602043152,"train_loss_before":0.6576848030090332}`

Both predeclared gates passed: train BCE decreased and held-out accuracy reached >=0.80. One invocation; 100 optimizer updates; no retry. This validates only deterministic local CUDA training feasibility for the tiny synthetic kernel—not parent #3807's five-seed LoRA hypothesis, which remains a separate allocation.

**Comment by Unjuno** [comment]
## Formal preflight result — PASS_CUDA_TRAINING_SMOKE

Frozen source SHA-256 matched exactly: `3f543c969f56a3ad030112afd35c865b6143109bfbfd9526f16026ec96a03c3b`. One invocation, exit code 0, Python 3.11 / PyTorch 2.5.1+cu121 / CUDA 12.1, NVIDIA GeForce RTX 3080 Laptop GPU, seed 3807, 100 steps, process-scoped `CUBLAS_WORKSPACE_CONFIG=:4096:8`, TF32 off, deterministic algorithms enabled.

Exact stdout JSON:

```json
{"cuda": "12.1", "device": "NVIDIA GeForce RTX 3080 Laptop GPU", "heldout_accuracy_after": 0.9921875, "heldout_accuracy_before": 0.5859375, "seed": 3807, "steps": 100, "train_loss_after": 0.14097687602043152, "train_loss_before": 0.6576848030090332}
```

Decision: `PASS_CUDA_TRAINING_SMOKE` (training BCE decreased; held-out accuracy increased above 0.80). This is an in-memory setup smoke probe only. No local files/checkpoints were created, and no PR/runtime integration is claimed. The separate five-seed rank-4 online LoRA allocation in parent #3807 remains unstarted and unchanged.
